python/graphstorm/model/gnn_with_reconstruct.py

```diff
@@ -61,7 +61,10 @@ def construct_node_feat(g, rel_names, input_gnn, get_input_embeds, batch_size,
                                             partition_book=g.get_partition_book(),
                                             ntype=ntype, force_even=False)
         # We use all neighbors to reconstruct node features.
-        sampler = dgl.dataloading.MultiLayerNeighborSampler([-1], mask=edge_mask)
+        fanout = {rel_name: 0 for rel_name in g.canonical_etypes}
+        for rel_name in rel_names:
+            fanout[rel_name] = -1
```

Comment:
I am a little bit confused, in GSgnnNodeSemiSupDataLoader, we have construct_feat_fanout, but here we use full neighbors. Why not use construct_feat_fanout here?

Reply:
this function is for inference. you can consider they are two bug fixes. we can split the PR into two PRs.